xarray/core/dataset.py

```diff
@@ -5065,15 +5063,15 @@ def drop_isel(self: T_Dataset, indexers=None, **indexers_kwargs) -> T_Dataset:
 
     def drop_dims(
         self: T_Dataset,
-        drop_dims: Hashable | Iterable[Hashable],
+        drop_dims: str | Iterable[Hashable],
```

Comment:
If `Dims = Union[str, Iterable[Hashable]]` this one can be replaced as well.
I think the other ones will still be ok as  `dim: Dims = None` will implicitly add None.
```python
def test(a: str = None):
    reveal_type(a)  # note: Revealed type is "Union[builtins.str, None]"
```

Reply:
I think explicit is always better, especially when switching to sphinx to write the docstring types.

Reply:
```suggestion
        drop_dims: Dims = None
```
I think aligning all these dim inputs with a single Dims type is a bigger win. But if we're going the explicit route then maybe this function should have None as default?

Reply:
Should I do it in this PR still?

Reply:
It can be done in a different PR if you prefer that. This is already an improvement.

Reply:
Ok, then lets do it in a different PR :) I will  not forget about it.